File: 01gpio/04pwm/pwm_wpi_sw.js
var wpi = require('wiring-pi');
var pin = 18;

wpi.setup('gpio');
wpi.pinMode(pin, wpi.OUTPUT);
wpi.softPwmCreate(pin, 0, 100);

while(true){
	for(var i=0; i<=100; i++){
		wpi.softPwmWrite(pin, i);
		wpi.delay(10);
	}
	for(var i=100; i>=0; i--){
		wpi.softPwmWrite(pin, i);
		wpi.delay(10);
	}
			
}

File: 01gpio/04pwm/pwm_wpi_sw.py
import wiringpi as wpi
PIN_LED = 18

wpi.wiringPiSetupGpio()
wpi.pinMode(PIN_LED, wpi.OUTPUT)
wpi.softPwmCreate(PIN_LED, 0, 100)

while True:
	for i in range(0,101): 
		wpi.softPwmWrite(PIN_LED, i)
		wpi.delay(10)
	
	for i in range(100,-1, -1):
		wpi.softPwmWrite(PIN_LED, i)
		wpi.delay(10)
